subprojects/jvm-services/build.gradle.kts

```diff
@@ -6,13 +6,19 @@ description = "JVM invocation and inspection abstractions"
 
 dependencies {
     implementation(project(":base-services"))
+    implementation(project(":core-api"))
```

Comment:
All these additions grow the transitive set of dependencies a lot.
Is that OK?
Was this project not one we need to keep slim?

Reply:
I honestly don't know if this is "ok". The problem is that basics like `Provider` live in `core-api` and I didn't see it as reasonable to add an abstraction over them to be used within the suppliers. 

@marcphilipp Can you shed some light how TD is using the `jvm-services` project?

Reply:
Like this:
```java
JvmVersionDetector jvmVersionDetector = project.getGradle().getServices().get(JvmVersionDetector.class);
String majorVersion = jvmVersionDetector.getJavaVersion(executable).getMajorVersion();
```

Reply:
Thanks marc.

We'll go ahead and introduce the dependency. If we ever want to reuse `jvm-services` outside Gradle, we can revisit.